addons/delivery/models/delivery_carrier.py

```diff
@@ -47,10 +47,16 @@ class DeliveryCarrier(models.Model):
     # Internals for shipping providers #
     # -------------------------------- #
 
-    delivery_type = fields.Selection([('grid', 'Default Delivery')], string="Delivery Type", default='grid', required='True')
+    delivery_type = fields.Selection([('fixed', 'Fixed Price'), ('base_on_rule', 'Based on Rule')], string="Delivery Type", default='fixed', required='True')
 
     price = fields.Float(compute='get_price')
     available = fields.Boolean(compute='get_price')
+    country_ids = fields.Many2many('res.country', 'delivery_carrier_country_rel', 'carrier_id', 'country_id', 'Countries')
+    state_ids = fields.Many2many('res.country.state', 'delivery_carrier_state_rel', 'carrier_id', 'state_id', 'States')
+    zip_from = fields.Char('Start Zip')
+    zip_to = fields.Char('To Zip')
+    price_rule_ids = fields.One2many('delivery.price.rule', 'carrier_id', 'Pricing Rules', copy=True)
+    state = fields.Selection([('test', 'Testing'), ('production', 'Production')], 'State', default='test')
```

Comment:
Hello @bst-odoo ,
 if delivery type is 'fixed' or 'based on rule' then default state should be 'production' and statusbar should be invisible.  and 
if delivery type is "shipping provider(Fedex, UPS, etc..)" then default state should be 'testing' and 
statusbar should be visible and clickable.